core/lib/Thelia/Module/BaseModuleInterface.php

```diff
@@ -26,5 +26,7 @@ public function preDeactivation(ConnectionInterface $con = null);
 
     public function postDeactivation(ConnectionInterface $con = null);
 
+    public function update($currentVersion, $newVersion, ConnectionInterface $con = null);
```

Comment:
This method already exists, she is just missing in the interface BaseModuleInterface.

https://github.com/gillesbourgeat/thelia/blob/8ccd16d01e68aca05cd0cd9c1180a642006ceb0b/core/lib/Thelia/Module/BaseModule.php#L517-L527